### ISSUE TITLE
Support logging

### ISSUE THREAD

**Issue by whatyouhide** [whatyouhide/corsica#5]

**Comment by whatyouhide** [comment]
Hi @blacksails, I pushed a couple of new commits which include the logging functionality we discussed in #4. Would you be willing to try out the `master` branch and tell me what you think? If you think this solves the problem I can push a new version to hex.pm.

To use the `master` just list the `:corsica` dependency as

```elixir
[...,
 {:corsica, github: "whatyouhide/corsica"}]
```

and to use logging add

```elixir
config :corsica, log_level: :info
```

to your `config/config.exs`. Thanks for all the help! :)

**Comment by blacksails** [comment]
I made a few comments to the changes you made in 29024e15c996f61c2b9edef2f1a77efcd41e27d8. Looks really good overall :) I will try out the code tomorrow.

**Comment by whatyouhide** [comment]
Hey @blacksails, thanks so much for the help! I pushed a bunch of commits which should move towards your suggested direction. Let me know what you think when you have a chance to try out the changes :)

**Comment by blacksails** [comment]
I tried out the logging. Works fine i think. You should bump the minor version (logging is new backwards compatible functionality). You can always make patch version bumps if we should find something's acting weird.

**Comment by whatyouhide** [comment]
Ok, I released `v0.3.0`. Thanks so much for the help and please don't hesitate to open new issues if the logging acts funny, it's super helpful! :)

**Comment by blacksails** [comment]
Happy to be of service :)
